.github/actions/download-artifacts/action.yml

```
@@ -0,0 +1,32 @@
+name: Download artifacts
+description: Download artifacts
+
+inputs:
+  path:
+    required: true
+    description: ""
+
+runs:
+  using: composite
+  steps:
+    - name: Download artifacts
+      uses: actions/download-artifact@v4
+      with:
+        path: ${{ github.workspace }}\DownloadBuildArtifacts
+
+    - name: Rename artifacts
+      shell: pwsh
+      env:
+        SOURCE_PATH: ${{ github.workspace }}\DownloadBuildArtifacts
+        TARGET_PATH: ${{ inputs.path }}
+      run: |
+        # Get list of artifacts files
+        $artifacts = Get-ChildItem -Path $env:SOURCE_PATH -Recurse | ForEach-Object { if (!($_.PSIsContainer)) { $_}} | Where-Object { $_.Name -like 'BusinessCentral.LinterCop*.dll' -and  $_.Directory.Name -like 'BusinessCentral.LinterCop*.dll' }
```

Also, what am I missing in the 2nd cond, why is there a .dll ending?

```suggestion
        $artifacts = Get-ChildItem -Path $env:SOURCE_PATH -Recurse -File
        | Where-Object { $_.Name -like 'BusinessCentral.LinterCop*.dll' -and  $_.Directory.Name -like 'BusinessCentral.LinterCop*.dll' }
```

---

The `actions/download-artifact@v4` download every artifact in it's own folder, where the name of the folder is the same as the artifact name.

The second condition is a extra safeguard that I only want to include these folders. In theory all the files in the `${{ github.workspace }}\DownloadBuildArtifacts` should be okay, this is just an extra precaution.